src/cloudflare/internal/pipeline-transform.ts

```diff
@@ -57,26 +58,39 @@ enum Format {
   JSON_STREAM = 'json_stream', // jsonl
 }
 
-export class PipelineTransformImpl extends entrypoints.WorkerEntrypoint {
+type PipelineBatchMetadata = {
+  pipelineId: string;
+  pipelineName: string;
+};
+
+type PipelineRecord = Record<string, unknown>;
```

Comment:
I assume our service is responsible for ensuring all records align with this specific format (i.e. of being an object with strings mapped to arbitrary values)? Because the JSON.parse() call in `readJsonStream` below won't guarantee that IIUC.

Reply:
Yeah exactly. If there is a problem here we need to address in the internal service.